### DISCUSSION TITLE
Support JsonResponse

### DISCUSSION THREAD

**Discussion by marcusr21** [dedoc/scramble#24] [Q&A] [answered]
Hey there! 

This is an incredible package that removes so much manual effort! 

We use `return new JsonResponse([])` in our controllers. This will include a `message`, `success` and `data` which relates to the Laravel Resource.

Is there any way we can support `\Illuminate\Http\JsonResponse`?

**Comment by romalytvynenko** [comment]
@marcusr21 hey! Yeah, this is possible.

Can you show an example of your real controller's action?

**Reply by marcusr21** [reply]
Thanks for replying so quick! 

```php
    public function show(Template $template)
    {
        return new JsonResponse([
            'success' => true,
            'message' => 'OK',
            'data' => TemplateResource::make($template)
        ]);
    }
```

**Reply by romalytvynenko** [reply]
What is inside `TemplateResource::make` method? Any chance this can be `new TemplateResource($template)` on your end?

**Reply by marcusr21** [reply]
```php
class TemplateResource extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array|\Illuminate\Contracts\Support\Arrayable|\JsonSerializable
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'description' => $this->description,
            'default_value' => $this->default_value,
            'company' => CompanyResource::make($this->whenLoaded('company')),
            'template_section' => TemplateSectionResource::make($this->whenLoaded('templateSection')),
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at
        ];
    }
}
```

Using `make` as it's a single instance, whereas if this was the `index` / GET route we would use `TemplateResource::collection`

**Reply by romalytvynenko** [reply]
Oh! I didn't know this is a part of Laravel 🫣

Will do!

Thank you!

**Comment by romalytvynenko** [comment, marked as answer]
Hey @marcusr21!

I've added `JsonResponse` support in `0.3.0`, please update: https://github.com/dedoc/scramble/releases/tag/v0.3.0

Thanks for the requesting the feature and sharing your case!
